src/App.tsx

```diff
@@ -1,14 +1,65 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodo, setCurrentTodo] = useState<Todo | undefined>();
+  const [currentUser, setCurrentUser] = useState<User | undefined>();
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState('all');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    let currentTodos = todos;
+    const formatedQuery = query.trim().toLowerCase();
+
+    if (filterType !== 'all') {
+      currentTodos = currentTodos.filter((todo) => {
+        const { completed } = todo;
+
+        if ((filterType === 'completed' && completed)
+          || (filterType === 'active' && !completed)) {
+          return true;
+        }
+
+        return false;
```

Comment:
I believe it should do the same
```suggestion
        return ((filterType === 'completed' && completed)
          || (filterType === 'active' && !completed)) 
```